rpc/client.go

```diff
@@ -2,25 +2,28 @@ package rpc
 
 import (
 	"bytes"
+	"context"
 	"encoding/json"
 	"fmt"
-	"io"
 	"net/http"
 )
 
-// JSONRPCCall executes a 2.0 JSON RPC HTTP Post Request to the provided URL with
+// JSONRPCCall calls JSONRPCCallWithContext with the default context
+func JSONRPCCall(url, method string, params ...interface{}) (Response, error) {
+	return JSONRPCCallWithContext(context.Background(), url, method, params...)
+}
+
+// JSONRPCCallWithContext executes a 2.0 JSON RPC HTTP Post Request to the provided URL with
 // the provided method and parameters, which is compatible with the Ethereum
 // JSON RPC Server.
-func JSONRPCCall(url, method string, parameters ...interface{}) (Response, error) {
-	const jsonRPCVersion = "2.0"
-
+func JSONRPCCallWithContext(ctx context.Context, url, method string, parameters ...interface{}) (Response, error) {
 	params, err := json.Marshal(parameters)
 	if err != nil {
 		return Response{}, err
 	}
 
 	req := Request{
-		JSONRPC: jsonRPCVersion,
+		JSONRPC: "2.0",
```

Comment:
Should we return the `jsonRPCVersion` constant?

Reply:
It is used in one place only so I just removed it and defined the value exactly where it is used.